content/en/docs/features/control-plane/entity-tag.md

```diff
@@ -0,0 +1,151 @@
+2. **Storage Failure**: If storing ETAG values fails, it can result in the loss of ETAG information and inconsistencies when determining resource modification. To mitigate this issue, the device developer should verify the return value from `oc_etag_dump` to confirm the success of the operation. Storage failures often occur due to insufficient storage space.
+3. **ETAG Reset**: In the event of a system restart without dump or power outage, the highest ETAG value stored in memory may be lost. To mitigate this, the ETAG values are not loaded through `oc_etag_load_and_clear`, requiring the client to resynchronize with the device.
+4. **Absence of Resources after Device Reboot**: ETAG values for non-existing resources are considered when calculating the highest ETAG value in `oc_etag_load_and_clear`. This ensures that ETAG values are not reused by new resources, avoiding conflicts and maintaining consistency.
+5. **Addition of Resources after Device Reboot**: Similar to creating a new resource, the ETAG value for a newly added resource is set either as the epoch time in milliseconds (`oc_clock_time()`) or as the highest ETAG value `oc_random_value()+1` if the epoch time is less than the current highest ETAG value. This ensures proper tracking of changes for the newly added resource.
+
+## Efficient Device Twin Synchronization
+
+In order to monitor resource changes and determine if a resource has been modified on the device, the CoAP gateway utilizes the Entity Tag (ETAG) mechanism. The Hub maintains the ETAG for each resource along with the timestamp of the last change. It is important to note that the timestamp refers to the time when the notification arrives at the CoAP gateway.
+
+For **Batch Observation**, the ETAG is associated with the overall state of resources. Prior to initiating resource observation, the CoAP gateway retrieves the latest ETAG among all device resources from the Hub Database. During resource observation initiation, the CoAP gateway sends the ETAG to the device. If the received ETAG matches the current state of the resources, the device responds with a code `VALID`. However, if the received ETAG does not match, the device responds with a code `CONTENT` and includes the current ETAG. Consequently, when a resource changes, the device sends the updated ETAG back to the CoAP gateway via a notification. The CoAP gateway transmits the ETAG together with the Content by using the `NotifyResourceChanged` method to the resource-aggregate. This command is then converted into a `ResourceChanged` event, which is saved in a database and distributed through the event bus. In cases where multiple resources change simultaneously, the CoAP gateway updates all affected resources with the same timestamp and ETAG.
```

Comment:
Could you please draw a sequence diagram for this process, between the coap gw and device? including the reconnect, updating of the resource, valid as well as content response, and loading from the db.

Reply:
Added